oxenss/rpc/client_rpc_endpoints.h

```diff
@@ -603,8 +615,117 @@ struct sequence : batch {
+///   (That is: conditions are "and"ed together).
+///
+/// - A `"then"` key contains a single request to invoke if the condition is satisfied.  The request
+///   itself is specified as a dict containing "method" and "params" keys containing the endpoint to
+///   invoke and the parameters to pass to the request.  The given request is permitted to be
+///   a nested "ifelse" or a "batch"/"sequence".  Note, however, that batch/sequence requests may
+///   not contain "ifelse" requests.
+///
+/// - An `"else"` key contains a single request to invoke if the condition is *not* satisfied.
+///   Parameters are the same as `"then"`.
+///
+/// `"if"` is always required, and at least one of "then" and "else" is required: if one or the
+/// other is omitted then no action is performed if that branch would be followed.
+///
+/// This endpoint returns a dict containing keys:
+/// - "hf" -- the current hardfork version (e.g. [19,1])
+/// - "v" -- the running storage server version (e.g. [2,3,0])
+/// - "height" -- the current blockchain height (e.g. 1234567)
+/// - "condition" -- true or false indicating the logical result of the `"if"` condition.
+/// - "result" -- a dict containing the result of the logical branch ("then" or "else") that was
+///   followed.  This dict has two keys:
+///   - "code" -- the numeric response code (e.g. 200 for a typical success)
+///   - "body" -- the response value (usually a dict).
+///   If the branch followed was omitted from the request (e.g. the condition failed and only a
+///   "then" branch was given) then this "result" key is omitted entirely.
+///
+/// Example:
+///
+/// Suppose HF 19.2 introduces some fancy new command "abcd" but earlier versions require executing
+/// a pair of commands "ab" and "cd" to get the same effect:
+///
+/// Request:
+///
+///     {
+///       "if": { "hf_at_least": [19,2] },
+///       "then": { "method": "abcd", "params": { "z": 1 } },
+///       "else": {
+///         "method": "batch",
+///         "params": {
+///           "requests": [
+///             {"method": "ab", "params": {"z": 1}},
+///             {"method": "cd", "params": {"z": 3}}
+///           ]
+///         }
+///       }
+///     }
+///
+/// If the 19.2 hf is active then the response would be:
+///
+///     {
+///       "hf": [19,2],
+///       "v": [2,3,1],
+///       "height": 1234567,
+///       "condition": true,
+///       "result": { "code": 200, "body": {"z_plus_4": 5}}
+///     }
+///
+/// Response from some blockchain height before hf 19.2:
+///
+///     {
+///       "hf": [19,1],
+///       "v": [2,3,1],
+///       "height": 1230000,
+///       "condition": false,
+///       "result": {
+///         "code": 200,
+///         "body": [
+///           {"code": 200, "body": {"z_plus_2": 3}},
+///           {"code": 200, "body": {"z_plus_2": 5}}
+///         ]
+///       }
+///     }
+///
+struct ifelse : endpoint {
```

Comment:
aw yea now all we need is a jump instruction and we will have our own turning complete json format.